modules/programs/i3status-rust.nix

```diff
@@ -258,8 +251,14 @@ in {
         '';
 
         source = settingsFormat.generate ("config-${cfgFileSuffix}.toml") ({
-          theme = cfg.theme;
-          icons = cfg.icons;
+          theme = if lib.versionAtLeast config.home.stateVersion "23.05" then {
```

Comment:
Is there a reason not to condition this on `cfg.package.version` instead?

Reply:
No really a strong reason, but I did this way:

- Because it was easier.
- This is just a small part of the migration process anyway. Most users will need to update their configuration, so I thought having the state version dictating this would make users more aware of the changes.

But if you think it is better, I can change the check to use the version instead.

Reply:
+1 for version.

Reply:
IMO this is not what stateVersion is for: users are not supposed to have to change it, ever (unless they're reinstalling their whole system). Configuration isn't state, we actually know statically which format to generate, so I would prefer using `lib.versionAtLeast cfg.package.version "0.30"`.

Reply:
Done.